src/Sulu/Component/Content/Metadata/PropertiesMetadata.php

```diff
@@ -111,6 +111,19 @@ private function removeSectionProperties(array $children)
                 continue;
             }
 
+            if ($child instanceof BlockMetadata) {
+                // deep cloning is necessary because the components must only be manipulated for these properties
```

Comment:
Still don't really understand why this is necessary :thinking:

Reply:
The problem is, that we need the properties with the `sections` for the `metadata` request and without the `sections` for the data-properties and it seems like both are referring to the same `BlockMetaData` object.

If we don't clone the `BlockMetaData` object and remove the `sections` then all the `sections` are also removed in the `metadata` request and the `section` is not rendered in the frontend. This occurs because the same object is used and thus should not be modified.

For the `sections` outside of a block this is not an issue because we don't have to modify an object, we just extract the inner properties and append them.
But to get rid of the `sections` inside of the `BlockMetaData` we have to modify the `components` directly on the object via the `$component->setChildren(..)` method.

Reply:
Ok, I see... Maybe write the comment then more like that? What do you think about this?

```php
// Clone before removing sections, because the metadata including the sessions is required elsewhere
```

Reply:
`... sessions ...` ?